### DISCUSSION TITLE
I would like to click on a table row and have a second table display the related data.

### DISCUSSION THREAD

**Discussion by pmankow** [Budibase/budibase#11542] [Help] [answered]
Hello.

I have two tables. Table Parts has a row Part Number, and another column with a relationship named Alternate to a second table named Alternates. Table Alternates has three columns, table one's Part Number (relationship to table Parts), a Supplier Part, and State. Any one row in table Parts can have many rows in table Alternates. 

In a screen view, when I click on a row in the Parts table, I would like the Alternates table to show the related rows.

How can I do this? I want this to be read only data that can be copied by the user or exported. I do not want it editable in this screen so no forms.

**Comment by ConorWebb96** [comment]
Hey @pmankow,

This can definitely be done, however will change slightly depending on the data source type you are using. For the demo I'll be going through I will be using the inbuilt internal Budibase data source. Furthermore, I will also be using the sample data provided within Budibase.

1. Setup 2 different data providers (rename them to prevent confusion e.g. parts provider, parts number provider)
2. Nest 2 different tables in each data provider
3. For your parts table click the `on row click` button.
<img width="100%" alt="Screenshot 2023-08-17 at 08 53 07" src="https://github.com/Budibase/budibase/assets/126772285/2fe7e638-7896-41f1-84e1-eed9ff17aa26">
4. Add a `fetch row` action, configure this so the table its using is the relationship table. Pass in the relationship id which should be stored in your relationship field within your parts table. `{{ Clicked row.Jobs.0._id }}` you will need to change jobs to part number or whatever the relationship name is.

5. Add a `update state` action, name it however you like, then click on the lighting bolt and assign a binding using the fetch row action setup previously. `{{ Action 1.Fetched row._id }}`
<img width="100%" alt="Screenshot 2023-08-17 at 09 00 49" src="https://github.com/Budibase/budibase/assets/126772285/7c86adb1-b900-41d7-bfe1-b6e78b79dece">
7. Go to your second data provider for part number.
8. Click define filters then set this up using the state created previously. Image below;
<img width="100%" alt="Screenshot 2023-08-17 at 09 01 52" src="https://github.com/Budibase/budibase/assets/126772285/397e25db-e74e-4bda-a2b9-af6f30b56d9d">

**End Result**
<img width="1418" alt="Screenshot 2023-08-17 at 09 03 43" src="https://github.com/Budibase/budibase/assets/126772285/7d96dcb7-19c2-43c3-91ad-22a946ebdee4">


That should be everything setup properly I hope this helps/points you in the right direction.

**Reply by pmankow** [reply]
@ConorWebb96 Thank you for the reply.
Following your instructions step by step with the sample data set, I cannot get this example to work. Here are my settings:
![Screenshot 2023-08-17 084903](https://github.com/Budibase/budibase/assets/21954921/2b2a6b6b-ca67-4055-ab86-061a5c0371a1)
![Screenshot 2023-08-17 085006](https://github.com/Budibase/budibase/assets/21954921/824b473f-fcfd-4538-851d-665d83459843)
![Screenshot 2023-08-17 085029](https://github.com/Budibase/budibase/assets/21954921/3dc3a3f4-9020-4672-b2c1-fd642529e801)
![Screenshot 2023-08-17 085055](https://github.com/Budibase/budibase/assets/21954921/13326800-2371-4d6b-992f-bc952b885408)

For me both tables remain fully populated and do not change or filter when clicked. I am trying this in design and preview modes.

The only setting I am unsure of is for the Fetch Row as there was no screenshot to copy.

Just a note, in your example, I would want the data from the EMPLOYEE LEVEL column to show in the second table when a row is clicked. I don't know if this distinction matters ultimately.

**Reply by pmankow** [reply]
@ConorWebb96 
I now have your example working by changing the Table setting in Fetch Row to Jobs. I believe I tried this earlier so don't know what the issue is, maybe the preview page wasn't fully loaded....

I will continue to try and get this working with my data (it still is not) but I am trying to implement the functionality that I noted at the end of my last response.

Thank you.

**Reply by pmankow** [reply]
It seems to be a long loading delay that was causing issues.

However, I cannot get multiple rows to show in the second table if I select an row such as Suzy in your first table. It should show rows for both Senoir and Manager in the second table.

**Reply by ConorWebb96** [reply]
Hey @pmankow,

I'm sorry to hear this isn't working for you.

Here is the fetch row screenshot; 
<img width="799" alt="Screenshot 2023-08-17 at 14 07 42" src="https://github.com/Budibase/budibase/assets/126772285/8dd3fe61-90d5-4dbc-8d1c-6e8520964bda">

Can you trying changing the limit on the second data provider to only 1?
<img width="304" alt="Screenshot 2023-08-17 at 14 04 11" src="https://github.com/Budibase/budibase/assets/126772285/bb27804d-4901-432f-85c9-86baad7b86a6">

Can you also possibly create a paragraph component and output the state you have created to it, just to verify the id.
<img width="1364" alt="Screenshot 2023-08-17 at 14 09 10" src="https://github.com/Budibase/budibase/assets/126772285/c513edcf-ecec-45bc-8ba7-8149497ccc52">

`Just a note, in your example, I would want the data from the EMPLOYEE LEVEL column to show in the second table when a row is clicked. I don't know if this distinction matters ultimately.`

Related to what you said on the above, that is quite a big difference, as the `Employee level` field isn't a relationship field. It is an options picker. 

Just to clarify, all you want is just for that specific field within the same table to be visible below? You likely wouldn't be able to use a table for this as, its a column within the same table. Instead you could output it with a different component e.g. paragraph etc.

If the above is true, you wouldn't need to fetch a row from a different table and instead save the options column and output it elsewhere.
<img width="1067" alt="Screenshot 2023-08-17 at 14 20 13" src="https://github.com/Budibase/budibase/assets/126772285/a4dcdd59-22f5-46e9-90fa-218952a3b9a5">
<img width="1367" alt="Screenshot 2023-08-17 at 14 20 55" src="https://github.com/Budibase/budibase/assets/126772285/854af7b9-7838-49ae-b3a6-43589de80a55">
<img width="1289" alt="Screenshot 2023-08-17 at 14 20 00" src="https://github.com/Budibase/budibase/assets/126772285/7372750c-8e39-4615-8bd4-ec66c69c4cbb">

**Reply by pmankow** [reply]
@ConorWebb96 
Sorry for the confusion. Here is my data:
-Part Number data/table:
![Screenshot 2023-08-17 092854](https://github.com/Budibase/budibase/assets/21954921/a1502b53-cbb6-41ab-9f21-e0294be70375)
-AML data/table:
![Screenshot 2023-08-17 093120](https://github.com/Budibase/budibase/assets/21954921/8e303d77-fd2e-440f-8ca8-371c5b9bc768)


From the above data being utilized in a setup like we have been working with above (two tables on a screen), to click a Part Number row and get the corresponding rows from the AML table. For instance, if they clicked the 0060-1003 Part Number row, the second table would display rows 1 & 2 from the AML data/table. That is what I meant in my original question, "In a screen view, when I click on a row in the Parts table, I would like the Alternates table to show the _related rows_."

Thanks.

**Reply by pmankow** [reply, marked as answer]
@ConorWebb96 I have figured this out.

1. I created two data providers with tables. Both data provider's data is Part data.

2. In the table under the first data provider (Data Provider 1); actions:
    a. Fetch Row; Table: Part Data 1, Row ID: {{ Clicked row._id }}
    b. Update State; Value: {{ State.Part_id }}

3. In the second data provider I filtered the Part data to equal {{ State.Part_id }}.

4. Under the second data provider (Data Provider 2) I then added a repeater with the conditions:
   a. Hide Component if {{ State.Part_id }} is empty. This prevents the screen from becoming a mess of empty tables.
6. Under the repeater I then added a third data provider: AML data.
    a. This data provider's data is New Repeater.Parts.AML

8. After the third data provider (AML data) I added a table. Here is the result:
![Screenshot 2023-08-17 122002](https://github.com/Budibase/budibase/assets/21954921/3875cdf9-f9f2-4488-8d27-03c588f9f482)

**Reply by ConorWebb96** [reply]
@pmankow happy you got it sorted! 

Thank you for the write up, I hope this helps someone else doing similar things if they stumble across this discussion! 🤝
